### ISSUE TITLE
The `val` method has issues with arrays

### ISSUE THREAD

**Issue by mwanago** [kysely-org/kysely#667]
I wrote the following query:
```
import { sql } from "kysely"

const data = {
  title: "hello",
  content: "world",
  categoryIds: [1, 2],
}

const authorId = 1

const databaseResponse = await kysely
  .with("created_article", (database) => {
    return database
      .insertInto("articles")
      .values({
        title: data.title,
        article_content: data.content,
        author_id: authorId,
      })
      .returningAll()
  })
  .insertInto("categories_articles")
  .columns(["article_id", "category_id"])
  .expression((expressionBuilder) => {
    return expressionBuilder
      .selectFrom("created_article")
      .select([
        "created_article.id as article_id",
        sql`unnest(${data.categoryIds}::int[]) as category_id`,
      ])
  })
  .returning((expressionBuilder) => {
    return [
      expressionBuilder
        .selectFrom("created_article")
        .select(["id", "title", "article_content", "author_id"]),
      expressionBuilder.val(data.categoryIds).as("category_ids"),
    ]
  })
  .executeTakeFirstOrThrow()
```
(you can find it on [kysely.link](https://kyse.link/?p=b&i=haFkqHBvc3RncmVzoXamMC4yNi4woXPaAktpbXBvcnQgeyBHZW5lcmF0ZWQgfSBmcm9tICJreXNlbHkiCgpkZWNsYXJlIGdsb2JhbCB7CiAgaW50ZXJmYWNlIENhdGVnb3JpZXNUYWJsZSB7CiAgICBpZDogR2VuZXJhdGVkPG51bWJlcj4KICAgIG5hbWU6IHN0cmluZwogIH0KCiAgaW50ZXJmYWNlIENhdGVnb3JpZXNBcnRpY2xlc1RhYmxlIHsKICAgIGNhdGVnb3J5X2lkOiBudW1iZXIKICAgIGFydGljbGVfaWQ6IG51bWJlcgogIH0KCiAgaW50ZXJmYWNlIEFydGljbGVzVGFibGUgewogICAgaWQ6IEdlbmVyYXRlZDxudW1iZXI-CiAgICB0aXRsZTogc3RyaW5nCiAgICBhcnRpY2xlX2NvbnRlbnQ6IHN0cmluZwogICAgYXV0aG9yX2lkOiBudW1iZXIKICB9CgogIGludGVyZmFjZSBEQiB7CiAgICBhcnRpY2xlczogQXJ0aWNsZXNUYWJsZQogICAgY2F0ZWdvcmllczogQ2F0ZWdvcmllc1RhYmxlCiAgICBjYXRlZ29yaWVzX2FydGljbGVzOiBDYXRlZ29yaWVzQXJ0aWNsZXNUYWJsZQogIH0KCiAgaW50ZXJmYWNlIFVzZXJUYWJsZSB7CiAgICBpZDogR2VuZXJhdGVkPHN0cmluZz4KICAgIGZpcnN0X25hbWU6IHN0cmluZyB8IG51bGwKICAgIGxhc3RfbmFtZTogc3RyaW5nCiAgfQp9CqFx2gP3aW1wb3J0IHsgc3FsIH0gZnJvbSAia3lzZWx5IgoKY29uc3QgZGF0YSA9IHsKICB0aXRsZTogImhlbGxvIiwKICBjb250ZW50OiAid29ybGQiLAogIGNhdGVnb3J5SWRzOiBbMSwgMl0sCn0KCmNvbnN0IGF1dGhvcklkID0gMQoKY29uc3QgZGF0YWJhc2VSZXNwb25zZSA9IGF3YWl0IGt5c2VseQogIC53aXRoKCJjcmVhdGVkX2FydGljbGUiLCAoZGF0YWJhc2UpID0-IHsKICAgIHJldHVybiBkYXRhYmFzZQogICAgICAuaW5zZXJ0SW50bygiYXJ0aWNsZXMiKQogICAgICAudmFsdWVzKHsKICAgICAgICB0aXRsZTogZGF0YS50aXRsZSwKICAgICAgICBhcnRpY2xlX2NvbnRlbnQ6IGRhdGEuY29udGVudCwKICAgICAgICBhdXRob3JfaWQ6IGF1dGhvcklkLAogICAgICB9KQogICAgICAucmV0dXJuaW5nQWxsKCkKICB9KQogIC5pbnNlcnRJbnRvKCJjYXRlZ29yaWVzX2FydGljbGVzIikKICAuY29sdW1ucyhbImFydGljbGVfaWQiLCAiY2F0ZWdvcnlfaWQiXSkKICAuZXhwcmVzc2lvbigoZXhwcmVzc2lvbkJ1aWxkZXIpID0-IHsKICAgIHJldHVybiBleHByZXNzaW9uQnVpbGRlcgogICAgICAuc2VsZWN0RnJvbSgiY3JlYXRlZF9hcnRpY2xlIikKICAgICAgLnNlbGVjdChbCiAgICAgICAgImNyZWF0ZWRfYXJ0aWNsZS5pZCBhcyBhcnRpY2xlX2lkIiwKICAgICAgICBzcWxgdW5uZXN0KCR7ZGF0YS5jYXRlZ29yeUlkc306OmludFtdKSBhcyBjYXRlZ29yeV9pZGAsCiAgICAgIF0pCiAgfSkKICAucmV0dXJuaW5nKChleHByZXNzaW9uQnVpbGRlcikgPT4gewogICAgcmV0dXJuIFsKICAgICAgZXhwcmVzc2lvbkJ1aWxkZXIKICAgICAgICAuc2VsZWN0RnJvbSgiY3JlYXRlZF9hcnRpY2xlIikKICAgICAgICAuc2VsZWN0KFsiaWQiLCAidGl0bGUiLCAiYXJ0aWNsZV9jb250ZW50IiwgImF1dGhvcl9pZCJdKSwKICAgICAgZXhwcmVzc2lvbkJ1aWxkZXIudmFsKGRhdGEuY2F0ZWdvcnlJZHMpLmFzKCJjYXRlZ29yeV9pZHMiKSwKICAgIF0KICB9KQogIC5leGVjdXRlVGFrZUZpcnN0T3JUaHJvdygpCqFjww==))

It produced
```
RETURNING
  (
    SELECT
      "id",
      "title",
      "article_content",
      "author_id"
    FROM
      "created_article"
  ),
  ($5, $6) AS "category_ids"
```
but it should be
```
RETURNING
  (
    SELECT
      "id",
      "title",
      "article_content",
      "author_id"
    FROM
      "created_article"
  ),
  $5 AS "category_ids"
```
instead. Cheers!

**Comment by koskimas** [comment]
Which version are you using?

**Comment by koskimas** [comment]
Nope, not fixed in 0.26.3 either. I think this is a bug, but might have been a conscious decision. I'll investigate.

As shown in the [playground](https://kyse.link/?p=s&i=qbBFjboccloXapoD9wQ9), you can work around this using `sql.val(array)`.

**Comment by mwanago** [comment]
Yeah, I was using `0.26.3`. When I was attaching the URL of `kysely.link` it didn't yet support `0.26.3` I think.
I'm not blocked by this issue, I found a workaround. I though that it might be worth pointing out, though.
Thanks a lot for all of your good work, I think Kysely is fantastic. I really hope it will continue growing.
